File: 2024/March/3-18 452 Minimum Number of Arrows to Burst Balloons/BurstBalloons.java
import java.util.Arrays;

class Solution {
    public int findMinArrowShots(int[][] points) {
        Arrays.sort(points, (a, b) -> { return a[1] - b[1]; });
        
        int currentPairPointer = 0;
        int x = points[currentPairPointer][1];
        int arrowsThrown = 1;

        while(currentPairPointer < points.length) {
            if(x >= points[currentPairPointer][0] && x <= points[currentPairPointer][1]) {
                currentPairPointer++;
            }
            else {
                arrowsThrown++;
                x = points[currentPairPointer][1];
            }
        }


        return arrowsThrown;
    }
}

File: 2024/March/3-18 452 Minimum Number of Arrows to Burst Balloons/BurstBalloons.js
/**
 * @param {number[][]} points
 * @return {number}
 */
var findMinArrowShots = function(points) {
    points.sort((a, b) => (a[1] - b[1]));

    let currentRange = 0;
    let x = points[0][1];
    let arrowsThrown = 1;

    while(currentRange < points.length) {
        if(x >= points[currentRange][0]
        && x <= points[currentRange][1]) {
            currentRange++;
        }
        else {
            arrowsThrown++;
            x = points[currentRange][1];
        }
    }

    return arrowsThrown;
};